programs/clearing_house/src/math/amm_jit.rs

```diff
@@ -13,25 +14,83 @@ mod tests;
 // assumption: taker_baa will improve market balance (see orders.rs & amm_wants_to_make)
 pub fn calculate_jit_base_asset_amount(
     market: &PerpMarket,
-    taker_base_asset_amount: u64,
+    maker_base_asset_amount: u64,
+    auction_price: u128,
+    valid_oracle_price: Option<i128>,
+    taker_direction: PositionDirection,
 ) -> ClearingHouseResult<u64> {
-    // simple impl
-    // todo: dynamic on imbalance
-    let mut jit_base_asset_amount = standardize_base_asset_amount(
-        taker_base_asset_amount
-            .checked_div(2)
-            .ok_or_else(math_error!())?,
-        market.amm.order_step_size,
-    )?;
-
-    if jit_base_asset_amount != 0 {
-        jit_base_asset_amount =
-            calculate_clampped_jit_base_asset_amount(market, jit_base_asset_amount)?;
-
-        jit_base_asset_amount =
-            standardize_base_asset_amount(jit_base_asset_amount, market.amm.order_step_size)?;
+    // only take up to 50% of what the maker is making
+    let mut max_jit_amount = maker_base_asset_amount
+        .checked_div(2)
+        .ok_or_else(math_error!())?;
+
+    // check for wash trade
+    if let Some(oracle_price) = valid_oracle_price {
+        let oracle_price = cast_to_u128(oracle_price)?;
+
+        // maker taking a short below oracle = likely to be a wash
+        // so we want to take less than 50%
+        if taker_direction == PositionDirection::Long && auction_price < oracle_price {
+            max_jit_amount = max_jit_amount.checked_div(4).ok_or_else(math_error!())?
```

Comment:
can we make this /100 ?

feel like if its a wash trade we want to be taking much less

Reply:
we can increase it later if it seems ok